run/cifs_archive/write-archive-configs-to.sh

```diff
@@ -5,4 +5,8 @@ FILE_PATH="$1"
 (
   echo "username=$SHARE_USER"
   echo "password=$SHARE_PASSWORD"
+  if [ -n "$SHARE_DOMAIN" ]
```

Comment:
This will fail with `SHARE_DOMAIN: unbound variable` if the variable is not set.
Use `"${SHARE_DOMAIN+x}"` instead.

Reply:
Thanks a lot for reviewing the PR.

It works for me like this.

With -n ${var+x} the condition would be always true. As I understand it the +x is only needed for a negative (-z) condition to distinguish between an empty variable from a variable that is not set.

Examples:
```
var_set="bar";
var_set_empty="";

[ -n "$var_set" ] && echo "var is set and not empty"
# var is set and not empty
[ -n "$var_set_empty" ] && echo "var is set and not empty"
[ -n "$var_notset" ] && echo "var is set and notempty"

[ -n ${var_set+x} ] && echo "always true"
# always true
[ -n ${var_set_empty+x} ] && echo "always true"
# always true
[ -n ${var_notset+x} ] && echo "always true"
# always true

[ -z ${var_set+x} ] && echo "var is not set"
[ -z ${var_set_empty+x} ] && echo "var is not set"
[ -z ${var_notset+x} ] && echo "var is not set"
# var is not set

[ -z "$var_set" ] && echo "var is empty or not set"
[ -z "$var_set_empty" ] && echo "var is empty or not set"
# var is empty or not set
[ -z "$var_notset" ] && echo "var is empty or not set"
# var is empty or not set
```

Reply:
> With `-n ${var+x}` the condition would be always true. As I understand it the +x is only needed for a negative (-z) condition to distinguish between an empty variable from a variable that is not set.

All of the teslausb scripts run with the `-eu` option:
```
pi@teslausb2:~ $ PS1="parent: $ "
parent: $ bash -eu
pi@teslausb2:~ $ if [ -n "$SHARE_DOMAIN" ]; then echo set; else echo not set; fi
bash: SHARE_DOMAIN: unbound variable
parent: $ 
```
Here I set the prompt of my shell to "parent", to distinguish it from the shell I start with `bash -eu`.
Then I run the `if` command, and you see that it prints an error message and exits the shell (because of the `-eu` options that the shell was started with)
However with the `+x` construct:
```
parent: $ bash -eu
pi@teslausb2:~ $ if [ -n "${SHARE_DOMAIN+x}" ]; then echo set; else echo not set; fi
not set
pi@teslausb2:~ $ 
```
and with SHARE_DOMAIN actually set to something:
```
pi@teslausb2:~ $ SHARE_DOMAIN=workgroup
pi@teslausb2:~ $ if [ -n "${SHARE_DOMAIN+x}" ]; then echo set; else echo not set; fi
set
```
The difference between that code and your examples is that in your examples, the `var+x` part isn't quoted.
I.e. there's a functional difference between:
```
[ -n ${var_notset+x} ] && echo "always true"
```
and
```
[ -n "${var_notset+x}" ] && echo "always true"
```

Reply:
Thank you for the detailed explanation.